firedrake/adjoint/interpolate.py

```diff
@@ -0,0 +1,21 @@
+from firedrake.adjoint.blocks import InterpolateBlock
+from pyadjoint.tape import get_working_tape, stop_annotating, annotate_tape
+
+
+def annotate_interpolate(interpolate):
+    def wrapper(interpolator, *function, **kwargs):
+        annotate = annotate_tape(kwargs)
+        with stop_annotating():
+            output = interpolate(interpolator, *function, **kwargs)
+
+        if annotate:
+            sb_kwargs = InterpolateBlock.pop_kwargs(kwargs)
```

Comment:
This looks wrong: if there are any kwargs that need popping (there probably aren't at the moment), that should be done before we go into `interpolate`